liquigraph-core/src/main/java/org/liquigraph/core/io/ConditionExecutor.java

```diff
@@ -63,7 +61,7 @@ private boolean execute(Connection connection, String query) {
             resultSet.next();
             return resultSet.getBoolean("result");
         }
-        catch (SQLException | ClientException e) {
+        catch (SQLException e) {
```

Comment:
To avoid breaking Liquigraph users, you could do something dirty like:

```java
catch (Exception e) {
            if (e instanceof SQLException || instanceOf(e, "org.neo4j.driver.exceptions.ClientException")) {
                throw new ConditionExecutionException(...)
            }
            throw e;
        }
```
where `instanceOf` is like (or better, there might be gotchas with the proposed version):

```java
    private static boolean instanceOf(Exception e, String className) {
        Class<?> type = load(className);
        return type == null || type.isAssignableFrom(e.getClass());
    }

    private static Class<?> load(String className) {
        try {
            return Class.forName(className);
        }
        catch (ClassNotFoundException e) {
            return null;
        }
    }
```

Maybe this instanceOf can live in a dedicated utility class.

Reply:
This is obviously ugly, but it makes the initial smell (i.e. hard dependency on JDBC implementation) quite obvious now, at least.

Reply:
I just realized `ClientException` is from the Neo4j native Java driver, not the JDBC driver... so it's definitely something that must be removed...

Reply:
Cool.